ion/symboltoken.go

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ion
+
+var UnknownSid int64 = -1
+
+// A SymbolToken providing both the symbol text and the assigned symbol ID.
+// Symbol tokens may be interned into a SymbolTable.
+// A text=nil or sid=-1 value might indicate that such field is unknown in the contextual symbol table.
+type SymbolToken struct {
+	text           *string
```

Comment:
Ditto on `*string`[1] and exporting these if `NewSymbolToken` and the accessors aren't going to do anything but set/get them.

[1] Although here I suppose you want to be able to distinguish between a symbol whose text is known to be the empty string and a symbol whose text is not known because of a missing shared symbol table? Ugh, I once again wish we hadn't made the empty string a valid symbol. :/

Reply:
> Ditto on *string[1] and exporting these if NewSymbolToken and the accessors aren't going to do anything but set/get them.

Updated to export the struct attributes and removed the setters/getters.

> [1] Although here I suppose you want to be able to distinguish between a symbol whose text is known to be the empty string and a symbol whose text is not known because of a missing shared symbol table? Ugh, I once again wish we hadn't made the empty string a valid symbol. :/

Thats correct, we need nil string to represent as unknown text.